osmosis-labs/osmosis-frontend
# Publish Stage

## What is the purpose of the change



* Truncate decimals
* New wallet onboarding modal
* Fix SF APR for stable pools
* Staking work behind feature flag (not testing now)
* Merge main and frontier
* Unsafe IBC currency registrar
* Multithreaded swap router 

## Test Guide
* New wallet onboarding modal: can connect to all wallets, including via wallet connect. Closing the modal during connection (QR code), cancellation, etc all work smoothly.
* Merge main and frontier
  * Setting works in settings modal
  * With setting off searching in swap tool and assets page prompts user to enable setting via modal
* Unsafe IBC currency registrar
  * Ensure all assets are available (as we see on frontier.osmosis.zone) and are able to be deposited and withdrawn
* New swap router
  * Reliability
    * Works well with slow network connection (can simulate this in swap tool)
    * Works well after idling for a while (a few minutes)
    * When assets are selected from drawer menu, the new quote reliably loads
